ui/components/app/connected-sites-list/connected-sites-list.component.js

```diff
@@ -9,56 +9,53 @@ export default class ConnectedSitesList extends Component {
   };
 
   static propTypes = {
-    connectedDomains: PropTypes.arrayOf(
+    connectedSubjects: PropTypes.arrayOf(
       PropTypes.shape({
         name: PropTypes.string,
-        icon: PropTypes.string,
+        iconUrl: PropTypes.string,
         origin: PropTypes.string,
         host: PropTypes.string,
       }),
     ).isRequired,
     onDisconnect: PropTypes.func.isRequired,
-    domainHostCount: PropTypes.objectOf(PropTypes.number).isRequired,
   };
 
   render() {
-    const { connectedDomains, onDisconnect } = this.props;
+    const { connectedSubjects, onDisconnect } = this.props;
     const { t } = this.context;
 
     return (
       <main className="connected-sites-list__content-rows">
-        {connectedDomains.map((domain) => (
+        {connectedSubjects.map((subject) => (
           <div
-            key={domain.origin}
+            key={subject.origin}
             className="connected-sites-list__content-row"
           >
-            <div className="connected-sites-list__domain-info">
-              <SiteIcon icon={domain.icon} name={domain.name} size={32} />
+            <div className="connected-sites-list__subject-info">
+              <SiteIcon icon={subject.iconUrl} name={subject.name} size={32} />
               <span
-                className="connected-sites-list__domain-name"
-                title={domain.extensionId || domain.origin}
+                className="connected-sites-list__subject-name"
+                title={subject.extensionId || subject.origin}
               >
-                {this.getDomainDisplayName(domain)}
+                {this.getSubjectDisplayName(subject)}
               </span>
             </div>
             <i
               className="fas fa-trash-alt connected-sites-list__trash"
               title={t('disconnect')}
-              onClick={() => onDisconnect(domain.origin)}
+              onClick={() => onDisconnect(subject.origin)}
             />
           </div>
         ))}
       </main>
     );
   }
 
-  getDomainDisplayName(domain) {
-    if (domain.extensionId) {
+  getSubjectDisplayName(subject) {
+    if (subject.extensionId) {
       return this.context.t('externalExtension');
     }
 
-    return this.props.domainHostCount[domain.host] > 1
-      ? domain.origin
-      : stripHttpSchemes(domain.origin);
+    return stripHttpSchemes(subject.origin);
```

Comment:
The host count was added so that we couldn't have multiple entries that appear different but are the same. For example if we had the same origin listed with different protocols (`http` and `https`) they'd look identical here.

Instead we could only strip `https://`, leaving `http://` as-is. That might be better anyway, so the user can see clearly that their connection isn't encrypted.

As a further improvement, we should probably also leave `https//` there if there is a non-standard port set. Otherwise it could be misleading (e.g. `foo.com:80` might look innocuous, but `https://foo.com:80` is more obviously non-stardard).

Reply:
Took a stab at it here: 73cdc0be8

Reply:
Looks good!

Reply:
Updated: a736af520